### PR TITLE
CI: Remove Ubuntu 18.04 configurations

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Removes two CI configurations using Ubuntu 18.04. Removing these configurations is necessary for the project to use C++17.

- The 64-bit configuration, which tested against several external services (MySQL, Postgres, etc.) is replaced by an Ubuntu 22.04 configuration testing against the same services.
- The 32-bit configuration is replaced by an Alpine 32-bit configuration

## What are related issues/pull requests?

#8270

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: .github/workflows/ubuntu_22.04/test.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+(cd autotest && OGR_MYSQL_CONNECTION_STRING=mysql:test,user=root,password=passwd,port=33060,host=$IP $PYTEST ogr/ogr_mysql.py)
+# MariaDB 10.3.9
+(cd autotest && OGR_MYSQL_CONNECTION_STRING=mysql:test,user=root,password=passwd,port=33061,host=$IP $PYTEST ogr/ogr_mysql.py)
+
+# PostGIS tests
+(cd autotest && OGR_PG_CONNECTION_STRING="host=$IP port=25432 dbname=autotest user=docker password=docker" $PYTEST --capture=no -ra ogr/ogr_pg.py)
+(cd autotest && PGHOST="$IP" PGPORT=25432 PGUSER=docker PGPASSWORD=docker $PYTEST --capture=no -ra gdrivers/postgisraster.py)
+
+# MongoDB v3
+(cd autotest && MONGODBV3_TEST_PORT=27018 MONGODBV3_TEST_HOST=$IP $PYTEST ogr/ogr_mongodbv3.py)
```
Comment (rouault) [inline .github/workflows/ubuntu_22.04/test.sh:29]: This test is skipped

File: .github/workflows/ubuntu_22.04/test.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+(cd autotest && OGR_MSSQL_CONNECTION_STRING="MSSQL:server=$IP;database=TestDB;driver=ODBC Driver 17 for SQL Server;UID=SA;PWD=DummyPassw0rd" $PYTEST ogr/ogr_mssqlspatial.py)
```
Comment (rouault) [inline .github/workflows/ubuntu_22.04/test.sh:31]: This test is skipped

Comment (dbaston) [reply .github/workflows/ubuntu_22.04/test.sh:31]: Fixed https://github.com/OSGeo/gdal/actions/runs/6801267162/job/18491718610?pr=8680#step:16:2684

File: .github/workflows/ubuntu_22.04/test.sh (hunk 1 above)
Comment (dbaston) [reply .github/workflows/ubuntu_22.04/test.sh:29]: Fixed https://github.com/OSGeo/gdal/actions/runs/6801267162/job/18491718610?pr=8680#step:16:2654

**Pull request merged**